torch/csrc/Generator.cpp

```diff
@@ -54,10 +55,15 @@ static PyObject* THPGenerator_pynew(
   c10::DeviceType device_type = device.type();
   if (device_type == at::kCPU) {
     self->cdata = make_generator<CPUGeneratorImpl>();
-  } else {
+  } else if (device_type == at::kCUDA) {
     self->cdata = globalContext()
                       .getAcceleratorHooksInterface(device_type)
                       .getNewGenerator(device.index());
+  } else if (c10::impl::hasGenerator(device_type)) {
+    self->cdata = at::Generator(c10::impl::makeGenerator(device));
+  } else {
+    throw std::runtime_error("No generator available for device type: " +
+                             c10::toString(device_type));
```

Comment:
So, wo do not need to modify this one, all the accelerators except CPU will follow the same `AcceleratorHooksInterface` logic.